pkgs/applications/misc/visidata/default.nix

```diff
@@ -25,19 +33,59 @@ buildPythonApplication rec {
   };
 
   propagatedBuildInputs = [
+    # from visidata/requirements.txt
+    # packages not (yet) present in nixpkgs are commented
     dateutil
-    pyyaml
+    pandas
+    requests
+    lxml
     openpyxl
     xlrd
     h5py
-    fonttools
-    lxml
-    pandas
+    psycopg2
     pyshp
+    #mapbox-vector-tile
+    #pypng
+    fonttools
+    #sas7bdat
+    #xport
+    #savReaderWriter
+    pyyaml
+    #namestand
+    #datapackage
+    pdfminer
+    #tabula
+    vobject
+    tabulate
+    wcwidth
+    zstandard
     setuptools
   ] ++ lib.optionals withPcap [ dpkt dnslib ];
 
-  doCheck = false;
+  checkInputs = [
+    git
+  ];
+
+  checkPhase = ''
+    # disable some tests which require access to the network
+    rm tests/load-http.vd            # http
+    rm tests/graph-cursor-nosave.vd  # http
+    rm tests/messenger-nosave.vd     # dns
+
+    # disable some tests which expect Python == 3.6 (not our current version)
+    # see https://github.com/saulpw/visidata/issues/1014
+    rm tests/describe.vd
+    rm tests/describe-error.vd
+    rm tests/edit-type.vd
+
+    # tests use git to compare outputs to references
+    git init -b "test-reference"
+    git config user.name "nobody"; git config user.email "no@where"
+    git add .; git commit -m "test reference"
+
+    substituteInPlace dev/test.sh --replace "bin/vd" "$out/bin/vd"
```

Comment:
That will break cross compile.

Reply:
I'm not sure how to work around this: the test script needs to run the vd executable, but running the one in the source repository doesn't work because it's not wrapped to have the proper PYTHONPATH.
Any clue on how to do that without using the $out bin?

Reply:
Do we even run checkPhase in cross-builds? If so, maybe it should be conditionally disabled…

Reply:
We could disable checkPhase when cross compiling. Just wanted to point that out.

Reply:
Fixed and amended :+1: